### PR TITLE
Disable more flashing elements if a11y is on

### PR DESCRIPTION
**Pull request by AllyTally**
## Changes:

For some reason, the accessibility option that was meant to disable flashes doesn't disable ALL flashes, only screen flashes and screen shaking. This commit disables a lot more, most importantly randomness in colors, the player flashing on death/respawn, and teleporters flashing.

## Legal Stuff:

By submitting this pull request, I confirm that...

- [x] My changes may be used in a future commercial release of VVVVVV
- [x] I will be credited in a `CONTRIBUTORS` file and the "GitHub Friends"
  section of the credits for all of said releases, but will NOT be compensated
  for these changes

### CODE REVIEW

File: desktop_version/src/Game.cpp [hunk 1]
```diff
@@ -400,7 +400,7 @@ void Game::lifesequence(void)
         if (lifeseq > 5) gravitycontrol = savegc;
 
         lifeseq--;
-        if (INBOUNDS_VEC(i, obj.entities) && lifeseq <= 0)
+        if (INBOUNDS_VEC(i, obj.entities) && lifeseq <= 0 || noflashingmode)
```
Comment (InfoTeddy) [inline desktop_version/src/Game.cpp:403]: This line needs to have parentheses to visually disambiguate which conditions are being AND-ed and OR-ed.

**Review by InfoTeddy — CHANGES_REQUESTED** [review]

File: desktop_version/src/Render.cpp [hunk 2]
```diff
@@ -2479,11 +2479,11 @@ static void rendermapcursor(const bool flashing)
         return;
     }
 
-    if (!flashing || (map.cursorstate == 2 && int(map.cursordelay / 15) % 2 == 0))
+    if (!flashing || (map.cursorstate == 2 && int(map.cursordelay / 15) % 2 == 0 || game.noflashingmode))
```
Comment (InfoTeddy) [inline desktop_version/src/Render.cpp:2482]:
```c++
VVVVVV/desktop_version/src/Render.cpp: In function ‘void rendermapcursor(bool)’:
VVVVVV/desktop_version/src/Render.cpp:2482:44: warning: suggest parentheses around ‘&&’ within ‘||’ [-Wparentheses]
 2482 |     if (!flashing || (map.cursorstate == 2 && int(map.cursordelay / 15) % 2 == 0 || game.noflashingmode))
      |                       ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: desktop_version/src/Render.cpp [hunk 3]
```diff
@@ -3041,7 +3041,7 @@ void teleporterrender(void)
 
     // Highlight the currently selected teleporter
 
-    if (game.useteleporter && help.slowsine % 16 > 8)
+    if (game.useteleporter && help.slowsine % 16 > 8 || game.noflashingmode)
```
Comment (InfoTeddy) [inline desktop_version/src/Render.cpp:3044]:
```c++
VVVVVV/desktop_version/src/Render.cpp: In function ‘void teleporterrender()’:
VVVVVV/desktop_version/src/Render.cpp:3044:28: warning: suggest parentheses around ‘&&’ within ‘||’ [-Wparentheses]
 3044 |     if (game.useteleporter && help.slowsine % 16 > 8 || game.noflashingmode)
      |         ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
```

File: desktop_version/src/Graphics.cpp [hunk 4]
```diff
@@ -3063,26 +3081,25 @@ void Graphics::menuoffrender(void)
 
 SDL_Color Graphics::huetilegetcol(const int t)
```
Comment (InfoTeddy) [inline desktop_version/src/Graphics.cpp:3082]:
```c++
VVVVVV/desktop_version/src/Graphics.cpp: In member function ‘SDL_Color Graphics::huetilegetcol(int)’:
VVVVVV/desktop_version/src/Graphics.cpp:3082:45: warning: unused parameter ‘t’ [-Wunused-parameter]
 3082 | SDL_Color Graphics::huetilegetcol(const int t)
      |                                   ~~~~~~~~~~^
```

**Review by InfoTeddy — CHANGES_REQUESTED** [review]

**Review by InfoTeddy — CHANGES_REQUESTED** [review]
If the teleporter is touched with screen effects on, its sprite will not animate.

**Comment by AllyTally** [comment]
Resolved and force pushed.

**Review by InfoTeddy — APPROVED** [review]

**Pull request merged**